source/uwp/Renderer/idl/AdaptiveCards.Uwp.idl

```diff
@@ -1440,14 +1440,15 @@ namespace AdaptiveCards
         interface IAdaptiveActionElement : IInspectable
         {
             [propget] HRESULT ActionType([out, retval] ActionType* value);
+            [propget] HRESULT ActionTypeString([out, retval] HSTRING* value);
 
             [propget] HRESULT Title([out, retval] HSTRING* value);
             [propput] HRESULT Title([in] HSTRING value);
 
             [propget] HRESULT Id([out, retval] HSTRING* value);
             [propput] HRESULT Id([in] HSTRING value);
 
-            [propget] HRESULT ActionTypeString([out, retval] HSTRING* value);
+            HRESULT GetAdditionalProperty([in] HSTRING propertyName, [out, retval] Windows.Data.Json.JsonObject** json);
```

Comment:
>GetAdditionalProperty [](start = 20, length = 21)

We discussed this offline. Instead of providing the GetAdditionalProperty where they need to specify the key they want to get, this can return a Json object that has all the additional properties. If all the caller cares about is passing the whole bag of additional properties somewhere else, they can do so, and if they just want to extract certain values, they can use the Windows.Data.Json apis to do it as well.

Reply:
I'm also going to make this a property now...

---
In reply to: [143250600](https://github.com/Microsoft/AdaptiveCards/pull/745#discussion_r143250600) [](ancestors = 143250600)

Reply:
Good catch